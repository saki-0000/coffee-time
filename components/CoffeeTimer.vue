<template>
  <Timer @change="time = $event">{{ label }}</Timer>
</template>

<script lang="ts">
import { computed, defineComponent, ref } from '@nuxtjs/composition-api'
import Timer from './ThreeMinutesTimer.vue'
import { useCoffeeTimer } from '~/composable/coffee-timer'

const { getLabel } = useCoffeeTimer()
export default defineComponent({
  components: { Timer },
  setup: () => {
    const time = ref(180)
    const label = computed(() => getLabel(time.value))
    return {
      time,
      label,
      getLabel,
    }
  },
})
</script>
