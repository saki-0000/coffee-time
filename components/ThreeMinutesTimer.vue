<template>
  <div>
    <v-btn id="startButton" :disabled="disabled" @click="onClick"
      ><slot>スタート</slot></v-btn
    >
    <span id="displayTime">
      {{ formatTime(time) }}
    </span>
  </div>
</template>

<script lang="ts">
import { defineComponent, ref, watchEffect } from '@nuxtjs/composition-api'
import { useTimer } from '~/composable/timer'

export default defineComponent({
  props: {
    second: {
      type: Number,
      default: 180,
    },
  },
  emits: ['change'],
  setup: (_, context) => {
    const { measureTime, time, formatTime } = useTimer()
    const disabled = ref(false)
    watchEffect(() => {
      context.emit('change', time.value)
    })
    const onClick = () => {
      disabled.value = true
      measureTime()
    }
    return {
      disabled,
      time,
      onClick,
      formatTime,
    }
  },
})
</script>
