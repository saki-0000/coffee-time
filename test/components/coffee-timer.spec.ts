import { shallowMount, Wrapper } from '@vue/test-utils'
import CoffeeTimer from '~/components/CoffeeTimer.vue'
import ThreeMinutesTimer from '~/components/ThreeMinutesTimer.vue'

jest.mock('~/composable/coffee-timer', () => ({
  useCoffeeTimer: () => ({
    getLabel: jest.fn(() => 'test'),
  }),
}))

describe('CoffeeTimer', () => {
  let wrapper: Wrapper<any>
  beforeEach(() => {
    wrapper = shallowMount(CoffeeTimer)
  })
  test('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  test('子コンポーネントからchangeイベントを受けたときにその返り値が表示されていること', () => {
    wrapper.findComponent(ThreeMinutesTimer).vm.$emit('change', 0)
    expect(wrapper.text()).toBe('test')
  })
})
