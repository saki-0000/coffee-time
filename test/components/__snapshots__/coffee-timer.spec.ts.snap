// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoffeeTimer renders correctly 1`] = `"<timer-stub second=\\"180\\">test</timer-stub>"`;
